File: tusow/src/main/java/it/unibo/coordination/tusow/exceptions/NotFoundError.java
package it.unibo.coordination.tusow.exceptions;

public class NotFoundError extends ClientError {

    private static final int CODE = 404;
    private static final String MSG = "Not Found";

    public NotFoundError() {
        super(CODE, MSG);
    }

    public NotFoundError(String message) {
        super(CODE, MSG + ": " + message);
    }

    public NotFoundError(String message, Throwable cause) {
        super(CODE, MSG + ": " +  message, cause);
    }

    public NotFoundError(Throwable cause) {
        super(CODE, MSG, cause);
    }
}

File: tusow/src/main/kotlin/it/unibo/coordination/tusow/exceptions/NotFoundError.kt
package it.unibo.coordination.tusow.exceptions

class NotFoundError : ClientError {
    constructor() : super(CODE, MSG)
    constructor(message: String) : super(CODE, "$MSG: $message")
    constructor(message: String, cause: Throwable?) : super(CODE, "$MSG: $message", cause)
    constructor(cause: Throwable?) : super(CODE, MSG, cause)

    companion object {
        private const val CODE = 404
        private const val MSG = "Not Found"
    }
}